src/app/databaseService.ts

```diff
@@ -0,0 +1,228 @@
+  databaseService: DatabaseServiceInterface,
+) {
+  const vehicleDetailsResult = vehicleDetailsQueryResult[0][0] as VehicleQueryResult;
+  const vehicleId = vehicleDetailsResult.id;
+  const vehicleDetails = vehicleDetailsResult.result;
+
+  vehicleDetails.technicalrecords = await getTechnicalRecords(vehicleId, databaseService);
+
+  return vehicleDetails;
+}
+
+async function getVehicleDetailsByVrm(
+  databaseService: DatabaseServiceInterface,
+  event: VehicleEvent,
+): Promise<VehicleDetails> {
+  console.info('Using get by VRM');
+  const vehicleDetailsQueryResult = await databaseService.get(technicalQueries.VEHICLE_DETAILS_VRM_QUERY, [
+    event.VehicleRegMark,
+  ]);
+
+  return getVehicleDetails(vehicleDetailsQueryResult, databaseService);
+}
+
+async function getVehicleDetailsByVin(
+  databaseService: DatabaseServiceInterface,
+  event: VehicleEvent,
+): Promise<VehicleDetails> {
+  console.info('Using get by VIN');
+  const vehicleDetailsQueryResult = await databaseService.get(technicalQueries.VEHICLE_DETAILS_VIN_QUERY, [
+    event.vinNumber,
+  ]);
+  return getVehicleDetails(vehicleDetailsQueryResult, databaseService);
+}
+
+async function getVehicleDetailsByTrailerId(
+  databaseService: DatabaseServiceInterface,
+  event: VehicleEvent,
+): Promise<VehicleDetails> {
+  console.info('Using get by Trailer ID');
+  const vehicleDetailsQueryResult = await databaseService.get(technicalQueries.VEHICLE_DETAILS_TRAILER_ID_QUERY, [
+    event.vinNumber,
+  ]);
+  return getVehicleDetails(vehicleDetailsQueryResult, databaseService);
+}
+
+async function hydrateTestRecord(
+  id: string,
+  testRecord: TestRecord,
+  databaseService: DatabaseServiceInterface,
+): Promise<TestRecord> {
+  const [[customDefects], [defects]] = await Promise.all([
+    databaseService.get(testQueries.CUSTOM_DEFECT_QUERY, [id]),
+    databaseService.get(testQueries.TEST_DEFECT_QUERY, [id]),
+  ]);
+
+  if (customDefects.length > 0) {
+    testRecord.customDefect = customDefects.map((customDefect: CustomDefectQueryResult) => customDefect.result);
+  } else {
+    testRecord.customDefect = [];
+  }
+
+  if (defects.length > 0) {
+    testRecord.defects = defects.map((defect: TestDefectQueryResult) => defect.result);
+  } else {
+    testRecord.defects = [];
+  }
+
+  return testRecord;
+}
+
+async function getTestRecordDetails(
+  queryResult: [RowDataPacket[], FieldPacket[]],
+  databaseService: DatabaseServiceInterface,
+): Promise<TestRecord> {
+  const testRecordResult = queryResult[0][0] as TestRecordQueryResult;
+  const testId = testRecordResult.id;
+  const testRecord = testRecordResult.result;
+
+  return hydrateTestRecord(testId, testRecord, databaseService);
+}
+
+async function getTestRecordsDetails(
+  queryResult: [RowDataPacket[], FieldPacket[]],
+  databaseService: DatabaseServiceInterface,
+): Promise<TestRecord[]> {
+  const testRecordResults = queryResult[0] as TestRecordQueryResult[];
+
+  return Promise.all(
```

Comment:
Do we need to wrap the map in a promise.all ? we use async which should already return a promise from the function. Wouldn't be the same if we just returned testRecordsResults.map(...)

Also for consistency across services I think it will be best if renamed testRecords to testResults perhaps?
This payload is shared across other services and as testResult  (cvs-svc-test-results) and tech-records as TechnicalRecord (cvs-svc-tech-records).
I am thinking that we were to troublebshoot, trace a payload across services, etc.. it might be easier if we try preserve some similar variable names.
What do you think?

Reply:
The `Promise.all` is so that the sub-requests are done concurrently, but the results aren't available until they've all finished. This should improve performance because we're not waiting for each DB request to finish before making the next one, we can kick them all off at once and only wait for the longest running one.

Renaming to testResults makes sense, I'll get on that now.

Reply:
Thanks for clarifying 👍